release 0.7.3

mvidalgarcia opened the pull request.

mvidalgarcia committed:
release: 0.7.3

mvidalgarcia committed:
validate: remove redundant success message

mvidalgarcia committed:
validate: show success msg if parameters are valid

tiborsimko commented on CHANGES.rst:
```diff
@@ -1,7 +1,7 @@
 =======
 
-Version 0.7.3 (UNRELEASED)
+Version 0.7.3 (2021-03-18)
```
03-24 or 03-25?

tiborsimko commented on CHANGES.rst:
```diff
@@ -1,7 +1,7 @@
+Version 0.7.3 (2021-03-18)
 --------------------------
 - Adds workflow parameters validation to ``validate`` command.
 - Adds option ``--environments`` to ``validate`` to perform workflow image validations.
```
WRT environment validation, one observation.  For BSM search, the output is huge, see below. This is because the same image is validated many times depending on where in the workflow it appears. Is there a change to quickly add a  sort-and-unique operation on extracted image names and tags, to have the simplest list to check, in order not to check the same image many times?

If it is fast, let's add it before 0.7.3 is out.  If it needs refactoring, let's create new issue for 0.7.4 (that could also have the #484 included).

```console
$ reana-demo-bsm-search
$ reana-client validate --environments
==> Verifying REANA specification file... /home/tibor/private/project/reana/src/reana-demo-bsm-search/reana.yaml
  -> SUCCESS: Valid REANA specification file.
==> Verifying workflow parameters and commands... 
  -> WARNING: Yadage input parameter "weightvariations" does not seem to be used.
  -> WARNING: Yadage input parameter "mcname" does not seem to be used.
  -> WARNING: Yadage input parameter "mcweight" does not seem to be used.
  -> WARNING: Yadage input parameter "shapevars" does not seem to be used.
==> Verifying environments in REANA specification file...
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists locally.
  -> SUCCESS: Environment image reanahub/reana-env-root6:6.18.04 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 has the correct format.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists locally.
  -> SUCCESS: Environment image reanahub/reana-demo-bsm-search:1.0.0 exists in Docker Hub.
  -> WARNING: Environment image UID is recommended to be 1000. UID 0 was found.
```

mvidalgarcia replied:
Workaround in https://github.com/reanahub/reana-client/pull/480/commits/a1975b889586a05a6a33c87ed62e66b94d404bd8. To be improved as part of https://github.com/reanahub/reana-client/issues/487.

tiborsimko commented on CHANGES.rst:
```diff
@@ -1,7 +1,7 @@
-Version 0.7.3 (UNRELEASED)
+Version 0.7.3 (2021-03-18)
 --------------------------
 - Adds workflow parameters validation to ``validate`` command.
```
Adds validation of workflow input parameters to the ``validate`` command.

tiborsimko replied:
Adds optional validation of workflow environment images to the ``validate`` command when `--environments`` option is used.

(hmm, a bit long... just wanted to find something similar to previous line, so that they would play nicely together...)

tiborsimko reviewed (approved):
Tested a few cases, works well.

Otherwise I tested a few cases, and all looks well WRT validation.


We may want to soften down the UID=0 message to say something like:

```
  -> INFO: Environment image uses UID=0 but will be run as UID=1000.
```

Can come afterwards

mvidalgarcia committed:
validate: keep track of validated environments

- avoids output repetition when multiple stages share same image
- improves performance as fewer API and docker calls occur
